internal/controller/helmchart_controller.go

```diff
@@ -665,7 +665,7 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 	// Construct the chart builder with scoped configuration
 	cb := chart.NewRemoteBuilder(chartRepo)
 	opts := chart.BuildOptions{
-		ValuesFiles:              obj.GetValuesFiles(),
+		ValuesFiles:              obj.Status.ObservedValuesFiles,
```

Comment:
This doesn't seem right… if building the chart, we should always take the list of requested ValuesFiles not the list that was found to exist during the previous build… Unless I've misunderstood, in the "first build" case, this is guaranteed to always discard explicitly specified `ValuesFiles` and instead pass as zero list. We need separate fields in `BuildOptions` for `ValuesFiles` vs `ObserviceValuesFiles`… no?

Reply:
Exactly.